x/bank/types/genesis.go

```diff
@@ -45,7 +46,21 @@ func ValidateGenesis(data GenesisState) error {
 		return err
 	}
 
-	return NewSupply(data.Supply).ValidateBasic()
+	totalSupply := NewSupply(data.Supply)
+	err := totalSupply.ValidateBasic()
+	if err != nil {
+		return err
+	}
+
+	var accsSupply sdk.Coins
+	for _, balance := range data.Balances {
+		accsSupply = append(accsSupply, balance.Coins...)
+	}
+
+	if !accsSupply.IsEqual(totalSupply.Total) {
+		return fmt.Errorf("total supply does not match with accounts balance")
+	}
+	return nil
```

Comment:
there's no need to add this because it is checked by the supply [invariant](https://github.com/cosmos/cosmos-sdk/blob/master/x/bank/keeper/invariants.go)